src/storage/hummock_sdk/src/lib.rs

```diff
@@ -320,23 +313,49 @@ impl EpochWithGap {
     }
 
     // return the epoch_with_gap(epoch + spill_offset)
-    pub(crate) fn from_u64(epoch_with_gap: u64) -> Self {
+    pub(crate) fn from_u64_real(epoch_with_gap: u64) -> Self {
         EpochWithGap(epoch_with_gap)
     }
 
     // return the pure epoch without spill offset
     pub fn pure_epoch(&self) -> HummockEpoch {
-        #[cfg(not(feature = "enable_test_epoch"))]
-        {
-            self.0 & !EPOCH_SPILL_TIME_MASK
-        }
-        #[cfg(feature = "enable_test_epoch")]
-        {
-            self.0
-        }
+        self.0 & !EPOCH_SPILL_TIME_MASK
     }
 
     pub fn offset(&self) -> u64 {
         self.0 & EPOCH_SPILL_TIME_MASK
     }
 }
+
+impl EpochWithGap {
```

Comment:
IIUC, the goal of this PR is to remove the usage of u64 epochs that are not correctly shifted in unit test and then replace the original invalid u64 value with a shifted one. Therefore, it's unnecessary to implement this logic via `EpochWithGap`, and we'd better keep `EpochWithGap` internal to hummock.

The invalid epochs are generated mostly in two ways, initializing from a random value (e.g. 1, 2, 233 etc.), and increment with a random value (+= 1, inc). 

For the first kind, in this PR, we have many code like `EpochWithGap::new_for_test(xxx).as_u64_for_test()` to turn a a random value into a well shifted value. This makes our code verbose, and also unnecessarily leak the concept of `EpochWithGap`. All it does is to convert an u64 to another u64, and we can implement it as a simple `fn test_epoch(value: u64) -> u64` to do the conversion.

For the second kind, for convenience to increment with usage `epochxxx.inc()`, we can use `east_ext` to implement some extended methods for u64 to increment to a next valid epoch like the following
```
#[easy_ext(NextEpoch)]
impl u64 {
    fn inc_epoch(&mut self) {
        xxx
    }

    fn next_epoch(self) -> u64 {
        xxx
    }
}
```
and then we can replace all usage of `<epoch: u64> += 1` and `<epoch: u64> + 1` with `epoch.inc_epoch()` and `epoch.next_epoch()`.

By now, the operation is all between u64, and no `EpochWithGap` is involved. Then we can move the implementation to `risingwave_common` and then we won't need import `risingwave_hummock_sdk` to get the correct epoch.

Reply:
Good idea!

Reply:
Good idea, I also feel that current code such as `EpochWithGap::new_for_test(xxx).as_u64_for_test()` is redundant, let me prioritize applying this comment.